code/modules/hivemind/mobs.dm

```diff
@@ -106,7 +106,7 @@
 
 /mob/living/simple_animal/hostile/hivemind/Life()
 	if(stat == DEAD)
-		return
+		return FALSE
```

Comment:
That's what I thought, but it didn't work that way in testing.

Reply:
I see where a return number may matter in processing, in case if 26 (PROCESS_KILL) is returned.
But there is no mob who does that... So what's the point? Mind sharing your testing results in more detail?

Reply:
The simple_animal Life() proc returns `TRUE` if nothing causes to return another value. I think `. = ..()` causes unspecified return values to be the same as `.`, which is `TRUE` when I step through the hivemind Life() proc.

Reply:
Living returns `TRUE` if the mob is not in stasis, the AI is active, and Life_Check() returns `TRUE`.

Reply:
And `Life_Check()` only `TRUE` if mob isn't dead. So Living won't return `TRUE` is mob is alive.
At least I don't see how would it.

Reply:
The deepmaint hivemind mobs are dying due to not having an existing hivemind AI. That check happens in hivemind Life() at line 129. When death() occurs at line 131, the proc continues to the child and does whatever behavior is defined there, which is causing a timer issue. This return might be fine as just `return`, but the one at line 132 needs to be `return FALSE`.

Reply:
I see how following 15% chance call of `malfunction()` may cause issues, if mob was gibbed.
That indeed could be avoided by swapping `death()` and `malfunction()` checks, or placing return after `death()` one.
That said, I don't get why return should be exactly `0`, but not `null`.

Reply:
I don't get it either. This was the only method that worked after a bunch of testing.

Reply:
Alright, fair enough. BYOND works in mysterious ways.